### PR TITLE
[core][spark]support format table create, and implement format table for SparkCatalog

### PR DESCRIPTION
**Pull request by adrian-wang**
### Purpose

Support format table creation, and add an implementation of format table for SparkCatalog.

### Tests

Added `SparkCatalogWithHiveTest`

### API and Format

No API changed.

### Documentation

https://cwiki.apache.org/confluence/display/PAIMON/PIP-27%3A+Introduce+Format+Table+in+Paimon+Catalog

### CODE REVIEW

**Review by JingsongLi — COMMENTED** [review]
Hi @adrian-wang , Thanks for the contribution!

1. Can you add documentation? Maybe in `sql-ddl`?
2. Can you add a test case? Maybe in `SparkGenericCatalogWithHiveTest`?

About how to create a format table, maybe we can just introduce a option `type`? Type = format?

**Comment by JingsongLi** [comment]
See https://github.com/apache/paimon/pull/4243

File: paimon-spark/paimon-spark-common/src/test/java/org/apache/paimon/spark/SparkCatalogWithHiveTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                                "spark.sql.catalog.spark_catalog.warehouse",
+                                warehousePath.toString())
+                        .master("local[2]")
+                        .getOrCreate();
+
+        spark.sql("CREATE DATABASE IF NOT EXISTS my_db1");
+        spark.sql("USE spark_catalog.my_db1");
+        spark.sql(
+                "CREATE TABLE IF NOT EXISTS table_orc (a INT, bb INT, c STRING) USING orc TBLPROPERTIES"
+                        + " ('paimon.format.table'='true')");
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/test/java/org/apache/paimon/spark/SparkCatalogWithHiveTest.java:69]: Can we avoid setting this option?

Comment (adrian-wang) [reply paimon-spark/paimon-spark-common/src/test/java/org/apache/paimon/spark/SparkCatalogWithHiveTest.java:69]: This is to identify a format table, with provider set as specified format.

Comment (adrian-wang) [reply paimon-spark/paimon-spark-common/src/test/java/org/apache/paimon/spark/SparkCatalogWithHiveTest.java:69]: If use `type` in #4243 , I feel like maybe it is not strong enough to tell generic catalog to delegate it to non-generic catalog

**Comment by adrian-wang** [comment]
@melin need to extend jdbc table def before support it.

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**